cloudformation/memsub-promotions-cf.json

```diff
@@ -0,0 +1,355 @@
+            ]
+          }
+        } ]
+      }
+    },
+
+    "InstanceProfile": {
+      "Type": "AWS::IAM::InstanceProfile",
+      "Properties": {
+        "Path": "/",
+        "Roles": [ {"Ref": "ServerRole"} ]
+      }
+    },
+
+    "LoadBalancer": {
+      "Type": "AWS::ElasticLoadBalancing::LoadBalancer",
+      "Properties": {
+        "Scheme" : "internet-facing",
+        "Listeners": [{
+          "LoadBalancerPort": "443",
+          "InstancePort": "9000",
+          "Protocol": "HTTPS",
+          "SSLCertificateId": {"Ref": "SslArn"}
+        }],
+        "CrossZone": "true",
+        "HealthCheck": {
+          "Target": "HTTP:9000/healthcheck",
+          "HealthyThreshold": "2",
+          "UnhealthyThreshold": "10",
+          "Interval": "30",
+          "Timeout": "10"
+        },
+        "Subnets": { "Ref": "PublicVpcSubnets" },
+        "SecurityGroups": [
+          { "Ref": "LoadBalancerSecurityGroup" }
+        ],
+        "ConnectionDrainingPolicy": {
+          "Enabled" : "true",
+          "Timeout" : "60"
+        }
+      }
+    },
+
+    "AutoscalingGroup": {
+      "Type": "AWS::AutoScaling::AutoScalingGroup",
+      "Properties": {
+        "AvailabilityZones": { "Fn::GetAZs": "" },
+        "LaunchConfigurationName": { "Ref": "LaunchConfig" },
+        "MinSize": {"Ref": "MinInstances"},
+        "MaxSize": {"Ref": "MaxInstances"},
+        "HealthCheckType": "ELB",
+        "HealthCheckGracePeriod": 400,
+        "LoadBalancerNames": [ {"Ref": "LoadBalancer"} ],
+        "Tags": [
+          {
+            "Key": "Stage",
+            "Value": { "Ref": "Stage" },
+            "PropagateAtLaunch": "true"
+          },
+          {
+            "Key": "Name",
+            "Value": {
+              "Fn::Join": [":", [{"Ref":"Stage"}, {"Ref":"App"}]]
+            },
+            "PropagateAtLaunch": "true"
+          },
+          {
+            "Key": "Stack",
+            "Value": {"Ref": "Stack"},
+            "PropagateAtLaunch": "true"
+          },
+          {
+            "Key": "App",
+            "Value": {"Ref": "App"},
+            "PropagateAtLaunch": "true"
+          },
+          {
+            "Key": "Role",
+            "Value": {"Ref": "App"},
+            "PropagateAtLaunch": "true"
+          }
+        ],
+        "VPCZoneIdentifier": {
+          "Ref": "PrivateVpcSubnets"
+        }
+      }
+    },
+
+    "LaunchConfig": {
+      "Type": "AWS::AutoScaling::LaunchConfiguration",
+      "Properties": {
+        "ImageId": {"Ref": "AmiId"},
+        "SecurityGroups": [
+          {"Ref": "InstanceSecurityGroup"},
+          {"Ref": "SshAccessSecurityGroup"}
+        ],
+        "InstanceType": {"Ref": "InstanceType"},
+        "AssociatePublicIpAddress": "True",
+        "IamInstanceProfile": {"Ref": "InstanceProfile"},
+        "UserData": {
+          "Fn::Base64": {
+              "Fn::Join": [
+                  "\n",
+                  [
+                      "#!/bin/bash -ev",
```

Comment:
Consider whether we want to do apt-get -y update and apt-get -y upgrade for this, or manage its AMI upgrades manually.